In [30]:
import pandas as pd
import os

silent = globals().get("silent", False)


### Daten laden

In [31]:
from config import DATA_PATH

df = pd.read_csv(DATA_PATH + "/diamonds.csv")
if not silent:
    display(df.head())

Unnamed: 0  carat      cut color clarity  depth  table  price     x     y  \
0           1   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98   
1           2   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84   
2           3   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07   
3           4   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23   
4           5   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35   

      z  
0  2.43  
1  2.31  
2  2.31  
3  2.63  
4  2.75

# Diamant-Datensatz Beschreibung

### Übersicht

Dieser Datensatz enthält Informationen über Diamanten, einschließlich ihrer physikalischen Eigenschaften und Preise. Der Datensatz besteht aus **53.940 Zeilen** und **10 Variablen** (Merkmalen). Hier ist eine detaillierte Beschreibung der einzelnen Merkmale:

### Variablen:

- **price**  
  Preis des Diamanten in US-Dollar (von \$326 bis \$18.823).

- **carat**  
  Gewicht des Diamanten (von 0,2 bis 5,01 Karat).

- **cut**  
  Qualität des Diamantschliffs. Die Kategorien umfassen:
  - Fair
  - Good
  - Very Good
  - Premium
  - Ideal

- **color**  
  Diamant-Farbgrad, der von J (schlechteste) bis D (beste) reicht.

- **clarity**  
  Ein Maß für die Reinheit des Diamanten, wobei niedrigere Werte mehr Einschlüsse und höhere Werte klarere Diamanten darstellen. Die Kategorien umfassen:
  - I1 (schlechteste)
  - SI2
  - SI1
  - VS2
  - VS1
  - VVS2
  - VVS1
  - IF (beste)

- **x**  
  Länge des Diamanten in Millimetern (von 0 bis 10,74 mm).

- **y**  
  Breite des Diamanten in Millimetern (von 0 bis 58,9 mm).

- **z**  
  Tiefe des Diamanten in Millimetern (von 0 bis 31,8 mm).

- **depth**  
  Gesamtprozent der Tiefe, berechnet als \( \text{depth} = \frac{2 \times z}{x + y} \). Werte reichen von 43% bis 79%.

- **table**  
  Die Breite der Oberseite des Diamanten relativ zu seinem breitesten Punkt, von 43% bis 95%.


In [ ]:
if not silent:
    print(f"Datensatz bevor er transformiert wurde:\n{df.dtypes.to_frame(name='Data Type')}")
    display(df.head())

# Zeige und lösche Spalten die nur NaN Werte enthalten
colsAllMissing = df.columns[df.isnull().all()]
if not silent:
    print(f"Der Datensatz enthaelt folgende Spalten die nie Werte haben: {colsAllMissing.values}")
df = df.drop(columns=colsAllMissing)

# Erste Spalte löschen, da es sich um einen ID im Dokument handelt
df = df.drop(columns=df.columns[0])

# Werte casten
df['cut'] = df['cut'].astype('string')
df['color'] = df['color'].astype('string')
df['clarity'] = df['clarity'].astype('string')


print(f"Finale Struktur des Datensatzes:\n{df.dtypes.to_frame(name='Data Type')}")




Before doing anything to the dataset:
           Data Type
Unnamed: 0     int64
carat        float64
cut           object
color         object
clarity       object
depth        float64
table        float64
price          int64
x            float64
y            float64
z            float64


Unnamed: 0  carat      cut color clarity  depth  table  price     x     y  \
0           1   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98   
1           2   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84   
2           3   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07   
3           4   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23   
4           5   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35   

      z  
0  2.43  
1  2.31  
2  2.31  
3  2.63  
4  2.75

The Data contains the following collumns which have no values: []
After deleting ID Row:
        Data Type
carat     float64
cut        object
color      object
clarity    object
depth     float64
table     float64
price       int64
x         float64
y         float64
z         float64


carat      cut color clarity  depth  table  price     x     y     z
0   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
1   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
2   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31
3   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23  2.63
4   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35  2.75

Final structure of the dataset:
              Data Type
carat           float64
cut              object
color    string[python]
clarity  string[python]
depth           float64
table           float64
price             int64
x               float64
y               float64
z               float64
